FORMAT.md

```diff
@@ -0,0 +1,226 @@
+# Imprint Binary Format Specification
+
+This document describes the binary format used by Imprint for serialization and deserialization, with ASCII art diagrams to illustrate the byte layout.
+
+## Record Structure Overview
+
+```text
++--------+-------------------------+---------+
+| Header | Field Directory (opt.)  | Payload |
```

Comment:
unrelated to this PR, but I think we should remove the optionality of the field directory. I initially thought it should be optional and you could pull it from a schema registry but I don't think that really makes sense anymore the way the design landed (you'd need an offset table anyway). 

Just leaving this comment to track so I'll create a ticket from it afterwards.